### ISSUE TITLE
running object-detection on cloud: "An error was raised. This may be due to a preemption in a connected worker or parameter server. The current session will be closed and a new session will be created. Error: OS Error"   

### ISSUE THREAD

**Issue by dori-reichmann** [tensorflow/models#5633]
I am trying to train object-detection following the tutuorial. When I run the same pipeline on a single machine, there are no problems.

The problem:
I constantly get the following Message: "An error was raised. This may be due to a preemption in a connected worker or parameter server. The current session will be closed and a new session will be created. Error: OS Error"
The message show up from all replicas which lead the to restart and waster precision time.

Here is a patch of the logs (I have hours of them...)

> I  worker-replica-1 Start master session 75a97c7043cae868 with config: allow_soft_placement: true graph_options { rewrite_options { meta_optimizer_iterations: ONE } } worker-replica-1 
> I  worker-replica-1 Graph was finalized. worker-replica-1 
> I  worker-replica-1 An error was raised. This may be due to a preemption in a connected worker or parameter server. The current session will be closed and a new session will be created. Error: OS Error worker-replica-1 
> I  worker-replica-4 Done running local_init_op. worker-replica-4 
> I  worker-replica-4 Running local_init_op. worker-replica-4 
> I  worker-replica-4 Start master session c68ba8523c26ae5c with config: allow_soft_placement: true graph_options { rewrite_options { meta_optimizer_iterations: ONE } } worker-replica-4 
> I  worker-replica-4 Graph was finalized. worker-replica-4 
> I  worker-replica-4 An error was raised. This may be due to a preemption in a connected worker or parameter server. The current session will be closed and a new session will be created. Error: OS Error worker-replica-4 
> I  worker-replica-0 Done running local_init_op. worker-replica-0 
> I  worker-replica-0 Running local_init_op. worker-replica-0 
> I  worker-replica-0 Start master session 4a156c81c7f04920 with config: allow_soft_placement: true graph_options { rewrite_options { meta_optimizer_iterations: ONE } } worker-replica-0 
> I  worker-replica-0 Graph was finalized. worker-replica-0 
> I  worker-replica-0 An error was raised. This may be due to a preemption in a connected worker or parameter server. The current session will be closed and a new session will be created. Error: OS Error worker-replica-0 
> I  worker-replica-0 loss = 1.4236093, step = 1084 (188.591 sec) worker-replica-0 
> I  worker-replica-4 Done running local_init_op. worker-replica-4 
> I  worker-replica-4 Running local_init_op. worker-replica-4 
> I  worker-replica-4 Start master session f8eab8afc462f35c with config: allow_soft_placement: true graph_options { rewrite_options { meta_optimizer_iterations: ONE } } worker-replica-4 
> I  worker-replica-4 Graph was finalized. worker-replica-4 
> I  worker-replica-4 An error was raised. This may be due to a preemption in a connected worker or parameter server. The current session will be closed and a new session will be created. Error: OS Error worker-replica-4 
> I  worker-replica-0 Done running local_init_op. worker-replica-0 
> I  worker-replica-0 Running local_init_op. worker-replica-0 
> I  worker-replica-0 Start master session 41207ae796d00daf with config: allow_soft_placement: true graph_options { rewrite_options { meta_optimizer_iterations: ONE } } worker-replica-0 
> I  worker-replica-0 Graph was finalized. worker-replica-0 
> I  worker-replica-0 An error was raised. This may be due to a preemption in a connected worker or parameter server. The current session will be closed and a new session will be created. Error: OS Error worker-replica-0 
> I  worker-replica-3 Done running local_init_op. worker-replica-3 
> I  worker-replica-3 Running local_init_op. worker-replica-3 
> I  worker-replica-3 Start master session bb761c1fbec37f9c with config: allow_soft_placement: true graph_options { rewrite_options { meta_optimizer_iterations: ONE } } worker-replica-3 
> I  worker-replica-3 Graph was finalized. worker-replica-3 
> I  worker-replica-3 An error was raised. This may be due to a preemption in a connected worker or parameter server. The current session will be closed and a new session will be created. Error: OS Error worker-replica-3 
> I  worker-replica-2 loss = 1.054275, step = 1018 (172.128 sec) worker-replica-2 
> I  worker-replica-3 loss = 0.97805524, step = 987 (175.943 sec) worker-replica-3 
> I  worker-replica-4 loss = 0.6143055, step = 987 (177.687 sec) worker-replica-4 
> I  worker-replica-0 Done running local_init_op. worker-replica-0 
> I  worker-replica-0 Running local_init_op. worker-replica-0 
> I  worker-replica-0 Start master session e6c64f34a170eaf3 with config: allow_soft_placement: true graph_options { rewrite_options { meta_optimizer_iterations: ONE } } worker-replica-0 
> I  worker-replica-1 loss = 0.65272343, step = 963 (159.242 sec) worker-replica-1 
> I  worker-replica-0 Graph was finalized. worker-replica-0 
> 

------------------------

### System information
-  What is the top-level directory of the model you are using: models/research/object-detection
-  Have I written custom code: no
- OS Platform and Distribution: N/A
-  Running on ML-engine (us-east1 and us-central1)
-  TensorFlow installed from: Conda
- Bazel version: N/A
- Tensorflow version = 1.9
- Python version = 2.,7
- **CUDA/cuDNN version**: ml-engine standard_gpu
- **GPU model and memory**: ml-engine standard_gpu
- **Exact command to reproduce**: Just following [](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/running_on_cloud.md)

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
Bazel version

**Comment by dori-reichmann** [comment]
@tensorflowbutler:  All fields are updated.

**Comment by wircho** [comment]
I'm experiencing this issue as well.

**Comment by pkulzc** [comment]
Are you using model_main.py? This seems to be a known issue with train.py which has been deprecated.

**Comment by wircho** [comment]
@pkulzc I'm not sure about @dori-reichmann, but I am using `model_main.py` and experiencing this issue.

It happens on all of the versions 1.10, 1.11, and 1.12.

**Comment by george1459** [comment]
I'm having this problem as well.

**Comment by Zhenyi-Liu** [comment]
Same error for me as well,@pkulz any thought?

**Comment by thoemi123** [comment]
Same error here also. However the error does not occure if I train a mobilenet model. For all other models I got the error. Training is still done but much slower than exptected (~2000 steps after 1 hour).

**Comment by pkulzc** [comment]
This looks like some CMLE issue about gRPC that happened once in the past. Could all users who have this issue share the following info?
1. your config
2. tf runtime
3. wether you synced to HEAD

**Comment by Zhenyi-Liu** [comment]
I used the example config: faster_rcnn_resnet101_kitti.config 
runtime: 1.9
I synced to HEAD

**Comment by thoemi123** [comment]
Config: faster_rcnn_resnet101_coco.config
runtime: 1.9
synced to HEAD: I don't know what this means (sorry I am quite new to TF/gcloud)

That said, I followed the tutorial (https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/running_pets.md) and used my own data/another config

**Comment by dori-reichmann** [comment]
My own config based on "faster_rcnn_resnet101_coco.config"
runtime: I tried 1.9, 1.10, 1.11
I was synced to HEAD when problem was observed (October 18)

**Comment by edumotya** [comment]
I think that it is not a object-detection problem, it is actually a CMLE issue. I am suffering from the same issue using a custom Estimator when distributing among workers (a single master machine always works). Increase the number of parameter servers should help, but it actually doesn't. More info at:

[https://stackoverflow.com/questions/52794441/encountering-preemption-os-error-when-trying-to-run-distributed-gcmle-job](url)

runtime: 1.12

**Comment by thoemi123** [comment]
I can confirm edumotya's answer. It does work when training on a single machine:

Cloud configuration:
trainingInput:
  runtimeVersion: "1.9"
  scaleTier: CUSTOM
  masterType: standard_gpu

**Comment by Zhenyi-Liu** [comment]
So there is no way to train on Multi-GPUs on using CMLE?

**Comment by pkulzc** [comment]
We ( object-detection and CMLE people) are investigating this now. 

Linking duplicated issues here:
https://github.com/tensorflow/models/issues/6220 
https://github.com/tensorflow/models/issues/4314

**Comment by george1459** [comment]
I can further confirm edumotya's answer. Multiple worker didn't work. Single GPU worked perfectly. Here's my cloud configuration:
trainingInput:
  runtimeVersion: "1.9"
  scaleTier: CUSTOM
  masterType: standard_gpu

**Comment by pkulzc** [comment]
This issue is gone in my test runs after https://github.com/tensorflow/models/pull/6315 is in. Could all of you in this thread sync your forks and re-run training?

**Comment by tomduval** [comment]
I am finding that this issue is still occurring on CMLE when going through this tutorial without changing anything:
https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/running_pets.md

Are there any ways to fix it?

**Comment by MirkoArnold1** [comment]
Same here. It is still occurring.
@tomduval for it did help for some models to increase the number of parameter servers, but if you have too many, you run into other issues. ssdlite_mobilenet_v2 trains fine for me with 9 workers and 3 parameter servers, while ssd_resnet50_v1_fpn training throws this error very often even with 6 parameter servers.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
